src/v/kafka/server/connection_context.h

```diff
@@ -399,6 +399,38 @@ class connection_context final
         ss::lowres_clock::time_point _last_request_timestamp;
     };
 
+    class throttling_state {
+    public:
+        ss::lowres_clock::duration update_fetch_delay(
+          ss::lowres_clock::duration new_delay,
+          ss::lowres_clock::time_point now) {
+            auto result_enforced = fetch_throttled_until - now;
+            fetch_throttled_until = now + new_delay;
+            return result_enforced;
```

Comment:
I guess this is more concise, but not clearer?
```suggestion
            return std::exchange(fetch_throttled_until, now + new_delay) - now;
```

Reply:
Yeah, with the `std::exchange` version I would have to expand it into the more verbose version in my mind.